app/exec/build-tasks-upload.ts

```diff
@@ -25,33 +27,55 @@ export class BuildTaskUpload extends cmdm.TfCommand {
     public exec(args: string[], options: cm.IOptions): any {
         var taskPath = args[0] || options['taskPath'];
         this.checkRequiredParameter(taskPath, 'taskPath', 'path to the task folder');
+        var deferred = Q.defer<agentifm.TaskDefinition>();
+        fs.readFile(path.join(taskPath, 'task.json'), 'utf8', (err, data) => {
+            if(err) {
+                return console.log(err);
+            }
 
-        var result: any = {};
-        result.task = { name: name };
-        var archive = archiver('zip');
-        archive.directory(taskPath, false);
-        //result.source = taskPath;
+            var taskJson = JSON.parse(data);
 
-        var deferred = Q.defer<agentifm.TaskDefinition>();
-        var agentapi = this.getWebApi().getTaskAgentApi(this.connection.accountUrl);
+            var vn = (taskJson.name  || taskPath);
 
-        agentapi.uploadTaskDefinition(archive, null, "", name, true, (err, statusCode, task) => {
-            if(err) {
-                err.statusCode = statusCode;
-                deferred.reject(err);
+            if (!taskJson.id || !check.isUUID(taskJson.id)) {
```

Comment:
breakout the validateTaskJson into another method at a minimum.  Ideally, put in a lib - but create a lib folder under exec - I only want commands in the direct files in exec